odoo/api.py

```diff
@@ -501,8 +501,12 @@ def __new__(cls, cr, uid, context, su=False):
         assert context is not None
         args = (cr, uid, context, su)
 
-        # if env already exists, return it
+        # determine transaction object
         transaction = cr.transaction
+        if transaction is None:
+            transaction = cr.transaction = Transaction(Registry(cr.dbname))
```

Comment:
This might be a bad suggestion. What is the most pythonic way to write this ?

```suggestion
        transaction = cr.transaction or Transaction(Registry(cr.dbname))
```

Reply:
The point is that if `cr.transaction is None`, you want to _reassign_ it.  So the "right" suggestion would be
```python
transaction = cr.transaction = cr.transaction or Transaction(Registry(cr.dbname))
```
The trick is that testing `value is None` is actually faster than testing `bool(value)`, which is what `or` does.
The code might look inelegant, but I think it does the minimum it has to do in every case, and it does it fast.
The creation of environments is a common operation in Odoo (think about `with_context`), and you want it to be as fast as possible.

Reply:
BTW the way multi-assignment works in Python
```python
a = b = <expr>
```
actually does
```python
a = <expr>
b = a
```

Reply:
I missed the multiple assignment on line 507... Thanks :)